Sources/SwiftParserDiagnostics/LexerDiagnosticMessages.swift

```diff
@@ -198,6 +202,15 @@ public extension SwiftSyntax.TokenDiagnostic {
       return [
         FixIt(message: .replaceCurlyQuoteByNormalQuote, changes: [[.replace(oldNode: Syntax(token), newNode: Syntax(fixedToken))]])
       ]
+    case .equalMustHaveConsistentWhitespaceOnBothSides:
+      let hasLeadingSpace = token.previousToken(viewMode: .all)?.trailingTrivia.contains(where: { $0.isSpaceOrTab }) ?? false
+      let fixedToken =
+        token
+        .with(\.leadingTrivia, hasLeadingSpace ? [] : .space)
+        .with(\.trailingTrivia, .space)
+      return [
+        FixIt(message: .insertWhitespace, changes: [[.replace(oldNode: Syntax(token), newNode: Syntax(fixedToken))]])
+      ]
```

Comment:
You should be able to use `replaceLeadingTrivia` and `replaceTrailingTrivia` instead of replacing the entire token. That should show up nicer in IDE’s which will then really only highlight the replaced trivia instead of also assuming that the token is replaced (by itself).

Also, I’d make sure that you only replace trivia if it’s really needed. I.e. if leading trivia already has a space, there’s no need to replace it.

---

But I agree that the entire `FixIt.Change` API is a mess and I want to improve that pretty soon-ish.